it/deployment_dependencies_test.go

```diff
@@ -0,0 +1,196 @@
+	deploymentDependenciesSecretName    = "secret1"
+)
+
+func TestConfigMapAndSecretToDeploymentToBundleIndex(t *testing.T) {
+	t.Parallel()
+
+	bundle := constructDeploymentDependenciesBundle()
+	SetupApp(t, bundle, false, true, assertConfigMapAndSecretToDeploymentToBundleIndex)
+}
+
+func assertConfigMapAndSecretToDeploymentToBundleIndex(ctx context.Context, t *testing.T, cfg *Config, args ...interface{}) {
+	// initial create of ConfigMap and Secret
+	cm, err := cfg.MainClient.CoreV1().ConfigMaps(cfg.Namespace).Create(deploymentDependenciesConfigMap())
+	require.NoError(cfg.T, err)
+	secret, err := cfg.MainClient.CoreV1().Secrets(cfg.Namespace).Create(deploymentDependenciesSecret())
+	require.NoError(cfg.T, err)
+
+	// Wait for stable state
+	deploymentName := cfg.Bundle.Spec.Resources[0].Spec.Object.(meta_v1.Object).GetName()
+	lw := cache.NewListWatchFromClient(cfg.MainClient.AppsV1().RESTClient(), "deployments", cfg.Namespace, fields.Everything())
+	cond := IsPodSpecAnnotationCond(t, cfg.Namespace, deploymentName, builtin.EnvRefHashAnnotation, "97b5461821487e047e760880764dd9b31a5b7f39e8961b98b809c6d3f5b3cf3a")
+	_, err = toolswatch.UntilWithSync(ctx, lw, &apps_v1.Deployment{}, nil, cond)
+	require.NoError(cfg.T, err)
+
+	// Update ConfigMap
+	cm.Data["b"] = "a"
+	_, err = cfg.MainClient.CoreV1().ConfigMaps(cfg.Namespace).Update(cm)
+	require.NoError(cfg.T, err)
+
+	// Wait for updated annotation
+	cond = IsPodSpecAnnotationCond(t, cfg.Namespace, deploymentName, builtin.EnvRefHashAnnotation, "5fb60505d125dd42160a9ef237873345020a90e8d607861ced50a641c3a800a9")
+	_, err = toolswatch.UntilWithSync(ctx, lw, &apps_v1.Deployment{}, nil, cond)
+	require.NoError(cfg.T, err)
+
+	// Update Secret
+	secret.StringData = map[string]string{
+		"z": "c",
+	}
+	_, err = cfg.MainClient.CoreV1().Secrets(cfg.Namespace).Update(secret)
+	require.NoError(cfg.T, err)
+
+	// Wait for updated annotation
+	cond = IsPodSpecAnnotationCond(t, cfg.Namespace, deploymentName, builtin.EnvRefHashAnnotation, "20ed86a5106aab7e892f498f38dc1b713405d864878e6a074377fbdcba129599")
+	_, err = toolswatch.UntilWithSync(ctx, lw, &apps_v1.Deployment{}, nil, cond)
+	require.NoError(cfg.T, err)
+}
+
+func constructDeploymentDependenciesBundle() *smith_v1.Bundle {
+	var (
+		replicas                      int32 = 1
+		minReadySeconds               int32 = 1
+		revisionHistoryLimit          int32 = 10
+		progressDeadlineSeconds       int32 = 5
+		terminationGracePeriodSeconds int64 = 30
+		labelMap                            = map[string]string{
+			"name": string(deploymentResourceName),
+		}
+	)
+	return &smith_v1.Bundle{
+		TypeMeta: meta_v1.TypeMeta{
+			Kind:       smith_v1.BundleResourceKind,
+			APIVersion: smith_v1.BundleResourceGroupVersion,
+		},
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name: "bundle-dt",
+		},
+		Spec: smith_v1.BundleSpec{
+			Resources: []smith_v1.Resource{
+				{
+					Name: deploymentResourceName,
+					Spec: smith_v1.ResourceSpec{
```

Comment:
We're going to have to make builders for this stuff or put it in files. I'm realizing there's.... a lot of boilerplate.